docs/extensions/intro.md

```diff
@@ -0,0 +1,170 @@
+    --> Created src/mynym/specterext/rubberduck/controller.py
+    --> Created src/mynym/specterext/rubberduck/config.py
+    --> Created src/mynym/specterext/rubberduck/__init__.py
+    --> Created src/mynym/specterext/rubberduck/__main__.py
+    --> Created src/mynym/specterext/rubberduck/templates/rubberduck/index.jinja
+    --> Created src/mynym/specterext/rubberduck/static/rubberduck/css/styles.css
+    --> Created src/mynym/specterext/rubberduck/static/rubberduck/img/ghost.png (via Github)
+    --> Created src/mynym/specterext/rubberduck/static/rubberduck/img/logo.jpeg (via Github)
+    --> Created src/mynym/specterext/rubberduck/templates/rubberduck/base.jinja
+    --> Created src/mynym/specterext/rubberduck/templates/rubberduck/transactions.jinja
+    --> Created src/mynym/specterext/rubberduck/templates/rubberduck/settings.jinja
+    --> Created src/mynym/specterext/rubberduck/templates/rubberduck/components/rubberduck_menu.jinja
+    --> Created src/mynym/specterext/rubberduck/templates/rubberduck/components/rubberduck_tab.jinja
+    --> Created pytest.ini
+    --> Created tests/conftest.py
+    --> Created tests/fix_ghost_machine.py
+    --> Created tests/fix_devices_and_wallets.py
+    --> Created tests/fix_testnet.py
+    --> Created tests/fix_keys_and_seeds.py
+    --> Created pyproject.toml
+    --> Created setup.py
+    --> Created setup.cfg
+    --> Created MANIFEST.in
+
+        Congratulations, you've created a new extension!
+
+        Here is how to get it to run in your development environment:
+            pip3 install -e .
+            python3 -m cryptoadvance.specter server --config DevelopmentConfig --debug
+            # Point your browser to http://localhost:25441
+            # Click "Choose plugins" --> rubberduck
+
+        If you want to package it, you can build it like this:
+            python3 -m pip install --upgrade build
+            python3 -m build
+            # Install it like this:
+            pip3 install dist/mynym_rubberduck-0.0.1-py3-none-any.whl
+
+        If you want to bring your extension to production, please refer to 
+        the readme in the dummy-extension repo:
+        https://github.com/cryptoadvance/specterext-dummy#how-to-get-this-to-production
+    
+        To publish your package
+
+            python3 -m pip install --upgrade twine
+            python3 -m twine upload --repository testpypi dist/*
+
+        You can get all this information again via:
+        python3 -m cryptoadvance.specter ext gen --help
+$ 
+```
+The created file structure looks like this and you will feel right at home if you have some knowledge about how Flask works:
+
+
+![](./images/extensions_file_layout.png)
+
+## Virtualenv Management
+
+The simplified example of creating an extension above is ommitting the use of a virtual environment. However, it's really __highly__ recommended that virtual environments are used for extension development. Without the Output above you would do something like:
```

Comment:
Since we're recommending virtual envs, wouldn't it make sense to move this before the non-VE approach?

Reply:
Yes, agreed. I've included the use of the virtual env directly in the TL;DR section.